apps/web/src/app/dataset/[dataset_id]/settings/basics/basic-form.tsx

```diff
@@ -18,61 +19,48 @@ import {
   FormMessage,
 } from '@/components/ui/form'
 import { Input } from '@/components/ui/input'
-import { FileProps } from '@/components/upload/utils'
+import { SchemaProps } from '@/app/dataset/type'
 
-import DocumentLoader, { stringUrlToFile } from './document-loader'
-import { SchemaProps } from './setting-page'
-import TextSplits from './splitter'
+import { FormSchema } from '../../../type'
 import TextEmbedding from './text-embedding'
-import { FormSchema } from './utils'
 import VectorStores from './vector-stores'
 
-interface IProps {
+export interface FormProps {
   datasetId?: string
-  showMore?: boolean
-  files?: FileProps[]
-  defaultValues: SchemaProps
-  setShowMore?: (s: boolean) => void
-  scrollRef: RefObject<HTMLDivElement>
-  sectionRefs: RefObject<HTMLDivElement>[]
+  name?: string
+  config?: any
   setUploading?: (s: boolean) => void
 }
 
 type Params = SchemaProps
 
 function editDataset(
   url: string,
-  { arg }: { arg: { name: string; config: Omit<Params, 'name'> } }
+  {
+    arg,
+  }: {
+    arg: { name: string; config: Omit<Params, 'name'> }
+  }
 ) {
   return fetcher(url, {
     method: 'PATCH',
     body: JSON.stringify(arg),
   })
 }
 
-const DatasetForm = ({
-  datasetId,
-  defaultValues,
-  setShowMore,
-  showMore,
-  scrollRef,
-  sectionRefs,
-  files,
-  setUploading,
-}: IProps) => {
-  const uploadFiles = useMemo(() => {
-    return files
-      ? files.reduce((m: FileProps[], item: FileProps) => {
-          const file = stringUrlToFile(item)
-          m?.push(file)
-          return m
-        }, [])
-      : []
-  }, [files])
-
-  const [data, setData] = useState<FileProps[]>(uploadFiles)
+const BasicsForm = ({ datasetId, config, name }: FormProps) => {
+  const defaultValues = useMemo(() => {
+    const basicsConfig = omit(config, [
+      'files',
+      'splitType',
+      'chunkSize',
+      'chunkOverlap',
+      'loaderType',
+    ])
+    return { name, ...basicsConfig }
+  }, [name, config])
+  // @ts-ignore
   const [values, setValues] = useState<SchemaProps>(defaultValues)
```

Comment:
```suggestion
  const defaultValues = useMemo(() => {
    const basicsConfig = omit(config, [
      'files',
      'splitType',
      'chunkSize',
      'chunkOverlap',
      'loaderType',
    ])
    return { name, ...basicsConfig } as SchemaProps
  }, [name, config])
  const [values, setValues] = useState<SchemaProps>(defaultValues)
```

这样就可以避免 ts-ignore 了。